oemof_b3/tools/testing_pipeline.py

```diff
@@ -0,0 +1,358 @@
+        Path to the current directory
+
+    Outputs
+    -------
+    target_path : str
+        Absolut path to target directory
+
+    """
+    target_path = current_path  # as the starting point
+
+    # Define the target directory name
+    target_directory = "oemof-B3"
+
+    # Loop until the target path is found
+    while os.path.basename(target_path) != target_directory:
+        target_path = os.path.dirname(target_path)
+        if target_path == os.path.expanduser("~"):
+            raise ValueError(
+                f"Target directory '{target_directory}' not found in the path hierarchy."
+            )
+
+    return target_path
+
+
+def rename_path(file_path, before, after):
+    """
+    This function renames existing files in directories by appending
+    the suffix "_original" to their filenames.
+
+    Inputs
+    -------
+    file_path : str
+        Absolute path to a directory
+    before : str
+        Original extension
+    after : str
+        Renamed extension
+
+    Outputs
+    -------
+    new_file_path : str
+        Renamed absolute file path
+
+    """
+    # Split the path and file name
+    directory, filename = os.path.split(file_path)
+
+    # Add suffix "_original" before the file extension
+    new_filename = filename.replace(before, after)
+
+    # Join the directory and new filename to get the new path
+    new_file_path = os.path.join(directory, new_filename)
+
+    # Check if file with new suffix already exists
+    if os.path.exists(new_file_path):
+        raise FileExistsError(
+            f"File {new_file_path} already exists and therefore we can not rename your file"
+            f" {filename}.\n"
+            f"Please rename the file {new_filename} first."
+        )
+
+    # Rename the file
+    os.rename(file_path, new_file_path)
+    print(f"File '{filename}' has been renamed.")
+
+    return new_file_path
+
+
+def get_raw_path():
+    this_path = os.path.abspath(os.getcwd())
+    repo_path = get_repo_path(this_path)
+    raw_dir_path = os.path.join(repo_path, "raw")
+
+    return raw_dir_path
+
+
+def check_raw_data_exists():
+    raw_dir_path = get_raw_path()
+
+    raw_dir_rule = ["raw/oemof-B3-raw-data.zip"]
+
+    if not os.path.isdir(raw_dir_path):
+        output = snakemake.snakemake(
+            targets=raw_dir_rule,
+            snakefile="Snakefile",
+        )
+
+        if not output:
+            raise FileExistsError(
+                f"The output corresponding to rule {raw_dir_rule[0]} could not be created. \n"
+                f"Hence this tests are failing."
+            )
+
+        return False
+    else:
+        return True
+
+
+def remove_raw_data_created(exists):
+    raw_dir_path = get_raw_path()
+
+    if not exists:
+        if os.path.isdir(raw_dir_path):
+            shutil.rmtree(raw_dir_path)
+        else:
+            raise FileNotFoundError(
+                f"Something went wrong. {raw_dir_path} has been created but could not be found."
+            )
```

Comment:
Could you please add docstrings to these @Stefanie08? Thank you!! :)

Reply:
The docstrings were added in the commit [4f93e74](https://github.com/rl-institut/oemof-B3/pull/357/commits/4f93e746157496b8da292f0ab1b828d56a611971).